extensions/sd_api_pictures/script.py

```diff
@@ -9,52 +10,92 @@
 import requests
 import torch
 from PIL import Image
+from datetime import date
 
 torch._C._jit_set_profiling_mode(False)
 
 # parameters which can be customized in settings.json of webui  
 params = {
-    'enable_SD_api': False,
-    'address': 'http://127.0.0.1:7860',
+    'address': '127.0.0.1:7860',
+    'mode': 0, # modes of operation: 0 (Manual only), 1 (Immersive/Interactive - looks for words to trigger), 2 (Picturebook Adventure - Always on)
+    'manage_VRAM': False,
     'save_img': False,
-    'SD_model': 'NeverEndingDream', # not really used right now
-    'prompt_prefix': '(Masterpiece:1.1), (solo:1.3), detailed, intricate, colorful',
+    'SD_model': 'NeverEndingDream', # not used right now
+    'prompt_prefix': '(Masterpiece:1.1), detailed, intricate, colorful',
     'negative_prompt': '(worst quality, low quality:1.3)',
-    'side_length': 512,
+    'width': 512,
+    'height': 512,
     'restore_faces': False
 }
 
+def update_params(): # somewhy the default extension params dict is not changed by settings.json, thus update it forcefully
+    ext_name = "sd_api_pictures"
+    for sett in shared.settings:
+        if sett.startswith(f'{ext_name}-'):
+            params.update({sett.replace(f'{ext_name}-',''): shared.settings[sett]})
+update_params()
+
+def give_VRAM_priority(actor):
+    
+    global shared, params
+
+    if (actor == 'SD'):
```

Comment:
Also minor, but I can see you're used to languages other than python - `()` on `if` isn't needed for python